packages/hoprd/src/admin.ts

```diff
@@ -81,19 +85,18 @@ export class AdminServer {
     this.wsServer = new ws.Server({ server: this.server })
 
     this.wsServer.on('connection', (socket: any, req: any) => {
-      if (!this.authenticate(req)) {
-        socket.send(
-          JSON.stringify({
-            type: 'auth-failed',
-            msg: 'authentication failed',
-            ts: new Date().toISOString()
-          })
-        )
-        socket.close()
-        return
-      }
-
       socket.on('message', (message: string) => {
+        if (!this.authenticate(req)) {
+          socket.send(
+            JSON.stringify({
+              type: 'auth-failed',
+              msg: 'authentication failed',
+              ts: new Date().toISOString()
+            })
+          )
+          return
```

Comment:
need to close socket

Reply:
Why?

Current solution produces flickering in the UI because the Websocket connection is continuously reestablished until the cookie is set.

The new version fails as well, but with `type: log, message: command rejected: authentication failed` instead of `type: 'auth-failed, message: authentication failed`